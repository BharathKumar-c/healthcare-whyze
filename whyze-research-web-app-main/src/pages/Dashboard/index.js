import React, { useEffect, useState } from 'react';
import { Text, Button } from '../../components/basic';
import './Dashboard.scss';
import {
  createFirstStudy,
  createFirstStudyContent,
  feasibilityStudiesInPlanning,
  seeAll,
} from '../../constant/ConstantTexts';
import ProjectLayout from './ProjectLayout';
import useDashboard from './useDashboard';
import { useSelector } from 'react-redux';
import SettingModal from '../../components/shared/SettingModal';
import { useDispatch } from 'react-redux';
import { clearSettings } from '../../redux/reducer/projectsReducer';
import { RoutesUrls } from '../../constant/RoutesUrls';
import { useNavigate } from 'react-router-dom';

const Dashboard = () => {
  const navigate = useNavigate();
  const { getProjects } = useDashboard();
  const { projects } = useSelector((state) => state.projectReducer);
  const { userDetails, isLoading } = useSelector((state) => state.authReducer);

  const dispatch = useDispatch();

  useEffect(() => {
    getProjects();
  }, []);
  const [openModal, setOpenModal] = useState(false);
  const handleClose = () => {
    setOpenModal(false);
    dispatch(clearSettings());
  };

  return (
    <div id="dashboard">
      <div className="dashboard">
        {!isLoading && (
          <div className="dashboard-container">
            <Text
              className="dashboard-title"
              text={`Welcome ${userDetails?.name ? userDetails?.name : ''}!`}
            />

            {projects?.length === 0 ? (
              <div className="dashboard-patient-study-container">
                <div className="dashboard-patient-study-text">
                  <div>
                    <Text className="dashboard-title" text={createFirstStudy} />
                  </div>
                  <div>
                    <Text
                      className="dashboard-patient-study-content"
                      text={createFirstStudyContent}
                    />
                  </div>
                </div>
                <div className="dashboard-patient-study-btn">
                  <Button
                    className="dashboard-patient-study-button"
                    label="Let's go"
                    handleButtonClick={() => setOpenModal(true)}
                  />
                </div>
              </div>
            ) : (
              <div className="dashboard-projects-list">
                <div className="dashboard-projects-list-header">
                  <div>
                    <Text
                      className="dashboard-projects-list-title"
                      text={feasibilityStudiesInPlanning}
                    />
                  </div>

                  <div
                    className="dashboard-projects-list-see-all-text"
                    onClick={() =>
                      navigate(`/${RoutesUrls.feasibilityStudiesInPlanning}`)
                    }
                  >
                    {seeAll}
                  </div>
                </div>

                <div className="dashboard-projects-container">
                  {projects?.length !== 0 &&
                    projects?.map((project, index) => (
                      <>
                        {index < 4 && (
                          <div
                            key={project.projectId}
                            className="dashboard-projects-container-content"
                          >
                            <ProjectLayout key={index} project={project} />
                          </div>
                        )}
                      </>
                    ))}
                </div>
              </div>
            )}
          </div>
        )}
      </div>
      {openModal && (
        <SettingModal
          openModal={openModal}
          handleClose={handleClose}
          type="create"
        />
      )}
    </div>
  );
};

export default Dashboard;
