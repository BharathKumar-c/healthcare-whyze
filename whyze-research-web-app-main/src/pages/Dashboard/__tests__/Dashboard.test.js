import React from 'react';
import {
  fireEvent,
  render,
  screen,
  act,
  waitFor,
} from '@testing-library/react';
import Dashboard from '..';
const mockedUsedNavigate = jest.fn();
import Store from '../../../redux/store';
import { Provider } from 'react-redux';
import {
  activeTrials,
  avgTimeLable,
  avgTimeToRecruit,
  countriesLable,
  createFirstStudy,
  createFirstStudyContent,
  patientstitleLable,
  screenFailRate,
  sitesLable,
  sitesRecruitedLable,
  threeMonths,
  totalPatientsLable,
  totalTrials,
  trialsLable,
  welComeComponentContent,
} from '../../../constant/ConstantTexts';
import { BrowserRouter } from 'react-router-dom';
import { setProjects } from '../../../redux/reducer/projectsReducer';
import ApiUtil from '../../../utils/ApiUtils';
import { dashboardApiUrls } from '../../../constant/ApiEndPoints';
import ProjectLayout from '../ProjectLayout';
import userEvent from '@testing-library/user-event';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUsedNavigate,
}));

jest.mock('axios');
jest.mock('../../../utils/ApiUtils');

function renderWithContext(element) {
  const { container, getByText } = render(
    <Provider store={Store}>{element}</Provider>,
  );
  return { container, Store, getByText };
}

describe('Test case for Dashbaord', () => {
  test('Dashboard render check', () => {
    const { container } = renderWithContext(<Dashboard />);
    expect(container).toMatchSnapshot();
  });
  test('Dashboard check UI components', () => {
    const { container } = renderWithContext(<Dashboard />);
    const patientsLogo = container.getElementsByClassName(
      'dashboard-patient-icon',
    );
    const dashboardTitle = container.getElementsByClassName('dashboard-title');
    const dashboardContentText = container.getElementsByClassName(
      'dashboard-content-text',
    );
    const patientTitle = container.getElementsByClassName(
      'dashboard-patient-text',
    );
    const totalPatients = container.getElementsByClassName(
      'dashboard-patient-total',
    );
    const totalPatientsText = container.getElementsByClassName(
      'dashboard-patient-total-text',
    );
    const countriesText = container.getElementsByClassName(
      'dashboard-patient-total-text',
    );
    const sitesLogo = container.getElementsByClassName('dashboard-sites-icon"');
    const sitesTitle = container.getElementsByClassName(
      'dashboard-patient-text',
    );
    const sitesRecruitedLableText = container.getElementsByClassName(
      'dashboard-patient-total-text',
    );
    const trialsTitle = container.getElementsByClassName(
      'dashboard-patient-text',
    );
    const trialsTimeTitle = container.getElementsByClassName(
      'dashboard-patient-total',
    );
    const trialsLableAndText = container.getElementsByClassName(
      'dashboard-patient-total-text',
    );
    const firstStudyTitle = container.getElementsByClassName('dashboard-title');
    const firstStudyTitleContent = container.getElementsByClassName(
      'dashboard-patient-study-content',
    );
    const letsgoButton = screen.getByRole('button');

    expect(patientsLogo).toBeDefined();
    expect(dashboardTitle.length).toBe(2);
    expect(dashboardContentText[0].textContent).toBe(welComeComponentContent);
    expect(patientTitle[0].textContent).toBe(patientstitleLable);
    expect(totalPatients[0].textContent).not.toBe('');
    expect(totalPatientsText[0].textContent).toBe(totalPatientsLable);
    expect(countriesText[1].textContent).toBe(countriesLable);
    expect(sitesLogo).toBeDefined();
    expect(sitesTitle[1].textContent).toBe(sitesLable);
    expect(sitesRecruitedLableText[2].textContent).toBe(sitesRecruitedLable);
    expect(sitesRecruitedLableText[3].textContent).toBe(avgTimeLable);
    expect(trialsTitle[2].textContent).toBe(trialsLable);
    expect(trialsLableAndText[4].textContent).toBe(activeTrials);
    expect(trialsLableAndText[5].textContent).toBe(totalTrials);
    expect(trialsLableAndText[6].textContent).toBe(screenFailRate);
    expect(trialsTimeTitle[7].textContent).toBe(threeMonths);
    expect(trialsLableAndText[7].textContent).toBe(avgTimeToRecruit);
    expect(firstStudyTitle[1].textContent).toBe(createFirstStudy);
    expect(firstStudyTitleContent[0].textContent).toBe(createFirstStudyContent);
    expect(letsgoButton).toBeInTheDocument();
  });
  test('Feasibility navigation', () => {
    renderWithContext(
      <BrowserRouter>
        <Dashboard />
      </BrowserRouter>,
    );
    expect(screen.getByRole('button')).toBeInTheDocument();
    fireEvent.click(screen.getByRole('button'));
    expect(mockedUsedNavigate).toHaveBeenCalledWith('/new-feasibility-study');
  });

  test('Update store to test rendering projects component', () => {
    const { Store } = renderWithContext(<Dashboard />);
    const projectData = [];

    act(() => {
      projectData.push({
        clientName: 'test',
        description: 'test test',
        feasibilityStudy: [
          {
            id: 'bfb1e873-755b-4ab7-9aa5-3ad4ec9a0dbb',
            projectId: 'e3653f73-e486-456b-9ba2-1e5421026175',
            isDone: false,
            feasibilityStudy: [
              {
                name: 'Indication',
                case: 'Indication',
                condition: 'Inclusion',
                caseName: 'Covid-19',
                child: [],
              },
              {
                name: 'Demographics',
                case: 'Demographics',
                condition: 'Inclusion',
                caseName: '',
                child: [
                  {
                    condition: 'Gender',
                    field: 'gender',
                    gender: '',
                  },
                  {
                    condition: 'Age range',
                    field: 'age',
                    ageRange: {
                      min: 0,
                      max: 0,
                    },
                  },
                ],
              },
            ],
            location: null,
            sitesContacted: null,
            sitesInitiated: null,
            preferredSites: null,
            sites: null,
            finalCount: 123,
          },
        ],
        id: 'e3653f73-e486-456b-9ba2-1e5421026175',
        isFavourite: false,
        patientSegmentReqiured: 123,
        projectName: 'test',
        userId: '0005a35c-b4b1-4fd4-bbd5-ab3e809e8b4e',
      });
      Store.dispatch(setProjects(projectData));
    });
  });
  test('Projects component', async () => {
    const { container, Store } = renderWithContext(<Dashboard />);

    const { projects } = JSON.parse(
      JSON.stringify(Store.getState().projectReducer),
    );
    expect(projects.length).toBe(1);

    const dashboardProjectComponent = container.getElementsByClassName(
      'dashboard-projects-list',
    );
    expect(dashboardProjectComponent).toHaveLength(1);
  });
});

describe('Test case for Use Dashbaord', () => {
  test('get Projects', async () => {
    renderWithContext(<Dashboard />);
    ApiUtil.postData.mockResolvedValue({
      data: [
        {
          clientName: 'test',
          description: 'test test',
          feasibilityStudy: [
            {
              id: 'bfb1e873-755b-4ab7-9aa5-3ad4ec9a0dbb',
              projectId: 'e3653f73-e486-456b-9ba2-1e5421026175',
              isDone: false,
              feasibilityStudy: [
                {
                  name: 'Indication',
                  case: 'Indication',
                  condition: 'Inclusion',
                  caseName: 'Covid-19',
                  child: [],
                },
                {
                  name: 'Demographics',
                  case: 'Demographics',
                  condition: 'Inclusion',
                  caseName: '',
                  child: [
                    {
                      condition: 'Gender',
                      field: 'gender',
                      gender: '',
                    },
                    {
                      condition: 'Age range',
                      field: 'age',
                      ageRange: {
                        min: 0,
                        max: 0,
                      },
                    },
                  ],
                },
              ],
              location: null,
              sitesContacted: null,
              sitesInitiated: null,
              preferredSites: null,
              sites: null,
              finalCount: 123,
            },
          ],
          id: 'e3653f73-e486-456b-9ba2-1e5421026175',
          isFavourite: false,
          patientSegmentReqiured: 123,
          projectName: 'test',
          userId: '0005a35c-b4b1-4fd4-bbd5-ab3e809e8b4e',
        },
        {
          clientName: 'test',
          description: 'test test',
          feasibilityStudy: [
            {
              id: 'bfb1e873-755b-4ab7-9aa5-3ad4ec9a0dbb',
              projectId: 'e3653f73-e486-456b-9ba2-1e5421026175',
              isDone: false,
              feasibilityStudy: [
                {
                  name: 'Indication',
                  case: 'Indication',
                  condition: 'Inclusion',
                  caseName: 'Covid-19',
                  child: [],
                },
                {
                  name: 'Demographics',
                  case: 'Demographics',
                  condition: 'Inclusion',
                  caseName: '',
                  child: [
                    {
                      condition: 'Gender',
                      field: 'gender',
                      gender: '',
                    },
                    {
                      condition: 'Age range',
                      field: 'age',
                      ageRange: {
                        min: 0,
                        max: 0,
                      },
                    },
                  ],
                },
              ],
              location: null,
              sitesContacted: null,
              sitesInitiated: null,
              preferredSites: null,
              sites: null,
              finalCount: 123,
            },
          ],
          id: 'e3653f73-e486-456b-9ba2-1e5421026175',
          isFavourite: false,
          patientSegmentReqiured: 123,
          projectName: 'test',
          userId: '0005a35c-b4b1-4fd4-bbd5-ab3e809e8b4e',
        },
        {
          clientName: 'test',
          description: 'test test',
          feasibilityStudy: [
            {
              id: 'bfb1e873-755b-4ab7-9aa5-3ad4ec9a0dbb',
              projectId: 'e3653f73-e486-456b-9ba2-1e5421026175',
              isDone: false,
              feasibilityStudy: [
                {
                  name: 'Indication',
                  case: 'Indication',
                  condition: 'Inclusion',
                  caseName: 'Covid-19',
                  child: [],
                },
                {
                  name: 'Demographics',
                  case: 'Demographics',
                  condition: 'Inclusion',
                  caseName: '',
                  child: [
                    {
                      condition: 'Gender',
                      field: 'gender',
                      gender: '',
                    },
                    {
                      condition: 'Age range',
                      field: 'age',
                      ageRange: {
                        min: 0,
                        max: 0,
                      },
                    },
                  ],
                },
              ],
              location: null,
              sitesContacted: null,
              sitesInitiated: null,
              preferredSites: null,
              sites: null,
              finalCount: 123,
            },
          ],
          id: 'e3653f73-e486-456b-9ba2-1e5421026175',
          isFavourite: false,
          patientSegmentReqiured: 123,
          projectName: 'test',
          userId: '0005a35c-b4b1-4fd4-bbd5-ab3e809e8b4e',
        },
        {
          clientName: 'test',
          description: 'test test',
          feasibilityStudy: [
            {
              id: 'bfb1e873-755b-4ab7-9aa5-3ad4ec9a0dbb',
              projectId: 'e3653f73-e486-456b-9ba2-1e5421026175',
              isDone: false,
              feasibilityStudy: [
                {
                  name: 'Indication',
                  case: 'Indication',
                  condition: 'Inclusion',
                  caseName: 'Covid-19',
                  child: [],
                },
                {
                  name: 'Demographics',
                  case: 'Demographics',
                  condition: 'Inclusion',
                  caseName: '',
                  child: [
                    {
                      condition: 'Gender',
                      field: 'gender',
                      gender: '',
                    },
                    {
                      condition: 'Age range',
                      field: 'age',
                      ageRange: {
                        min: 0,
                        max: 0,
                      },
                    },
                  ],
                },
              ],
              location: null,
              sitesContacted: null,
              sitesInitiated: null,
              preferredSites: null,
              sites: null,
              finalCount: 123,
            },
          ],
          id: 'e3653f73-e486-456b-9ba2-1e5421026175',
          isFavourite: false,
          patientSegmentReqiured: 123,
          projectName: 'test',
          userId: '0005a35c-b4b1-4fd4-bbd5-ab3e809e8b4e',
        },
        {
          clientName: 'test',
          description: 'test test',
          feasibilityStudy: [
            {
              id: 'bfb1e873-755b-4ab7-9aa5-3ad4ec9a0dbb',
              projectId: 'e3653f73-e486-456b-9ba2-1e5421026175',
              isDone: false,
              feasibilityStudy: [
                {
                  name: 'Indication',
                  case: 'Indication',
                  condition: 'Inclusion',
                  caseName: 'Covid-19',
                  child: [],
                },
                {
                  name: 'Demographics',
                  case: 'Demographics',
                  condition: 'Inclusion',
                  caseName: '',
                  child: [
                    {
                      condition: 'Gender',
                      field: 'gender',
                      gender: '',
                    },
                    {
                      condition: 'Age range',
                      field: 'age',
                      ageRange: {
                        min: 0,
                        max: 0,
                      },
                    },
                  ],
                },
              ],
              location: null,
              sitesContacted: null,
              sitesInitiated: null,
              preferredSites: null,
              sites: null,
              finalCount: 123,
            },
          ],
          id: 'e3653f73-e486-456b-9ba2-1e5421026175',
          isFavourite: false,
          patientSegmentReqiured: 123,
          projectName: 'test',
          userId: '0005a35c-b4b1-4fd4-bbd5-ab3e809e8b4e',
        },
        {
          clientName: 'test',
          description: 'test test',
          feasibilityStudy: [
            {
              id: 'bfb1e873-755b-4ab7-9aa5-3ad4ec9a0dbb',
              projectId: 'e3653f73-e486-456b-9ba2-1e5421026175',
              isDone: false,
              feasibilityStudy: [
                {
                  name: 'Indication',
                  case: 'Indication',
                  condition: 'Inclusion',
                  caseName: 'Covid-19',
                  child: [],
                },
                {
                  name: 'Demographics',
                  case: 'Demographics',
                  condition: 'Inclusion',
                  caseName: '',
                  child: [
                    {
                      condition: 'Gender',
                      field: 'gender',
                      gender: '',
                    },
                    {
                      condition: 'Age range',
                      field: 'age',
                      ageRange: {
                        min: 0,
                        max: 0,
                      },
                    },
                  ],
                },
              ],
              location: null,
              sitesContacted: null,
              sitesInitiated: null,
              preferredSites: null,
              sites: null,
              finalCount: 123,
            },
          ],
          id: 'e3653f73-e486-456b-9ba2-1e5421026175',
          isFavourite: false,
          patientSegmentReqiured: 123,
          projectName: 'test',
          userId: '0005a35c-b4b1-4fd4-bbd5-ab3e809e8b4e',
        },
      ],
    });
    await waitFor(() => {
      expect(ApiUtil.postData).toHaveBeenCalledWith(
        `${dashboardApiUrls.projectUrl}`,
        {
          userId: '0005a35c-b4b1-4fd4-bbd5-ab3e809e8b4e',
        },
      );
      expect(ApiUtil.postData).toHaveBeenCalledTimes(1);
      const { segments } = JSON.parse(
        JSON.stringify(Store.getState().feasibilityStudyReducer),
      );
      expect(segments.length).toBeGreaterThan(0);
    });
  });

  test('get Projects by id', async () => {
    const data = {
      clientName: 'test',
      description: 'test test',
      feasibilityStudy: [
        {
          id: 'bfb1e873-755b-4ab7-9aa5-3ad4ec9a0dbb',
          projectId: 'e3653f73-e486-456b-9ba2-1e5421026175',
          isDone: false,
          feasibilityStudy: [
            {
              name: 'Indication',
              case: 'Indication',
              condition: 'Inclusion',
              caseName: 'Covid-19',
              child: [],
            },
            {
              name: 'Demographics',
              case: 'Demographics',
              condition: 'Inclusion',
              caseName: '',
              child: [
                {
                  condition: 'Gender',
                  field: 'gender',
                  gender: '',
                },
                {
                  condition: 'Age range',
                  field: 'age',
                  ageRange: {
                    min: 0,
                    max: 0,
                  },
                },
              ],
            },
          ],
          location: null,
          sitesContacted: null,
          sitesInitiated: null,
          preferredSites: null,
          sites: null,
          finalCount: 123,
        },
      ],
      id: 'e3653f73-e486-456b-9ba2-1e5421026175',
      isFavourite: false,
      patientSegmentReqiured: 123,
      projectName: 'test',
      userId: '0005a35c-b4b1-4fd4-bbd5-ab3e809e8b4e',
    };
    const { container } = renderWithContext(
      <ProjectLayout project={data} currentIndex={0} />,
    );
    const projectTitle = container.getElementsByClassName(
      'project-layout-indication-title',
    );
    expect(projectTitle[0].textContent).toBe('Covid-19');
    fireEvent.click(projectTitle[0]);
    ApiUtil.getData.mockResolvedValue({
      data: {
        clientName: 'test',
        description: 'test test',
        feasibilityStudy: [
          {
            id: 'bfb1e873-755b-4ab7-9aa5-3ad4ec9a0dbb',
            projectId: 'e3653f73-e486-456b-9ba2-1e5421026175',
            isDone: false,
            feasibilityStudy: [
              {
                name: 'Indication',
                case: 'Indication',
                condition: 'Inclusion',
                caseName: 'Covid-19',
                child: [],
              },
              {
                name: 'Demographics',
                case: 'Demographics',
                condition: 'Inclusion',
                caseName: '',
                child: [
                  {
                    condition: 'Gender',
                    field: 'gender',
                    gender: '',
                  },
                  {
                    condition: 'Age range',
                    field: 'age',
                    ageRange: {
                      min: 0,
                      max: 0,
                    },
                  },
                ],
              },
            ],
            location: null,
            sitesContacted: null,
            sitesInitiated: null,
            preferredSites: null,
            sites: null,
            finalCount: 123,
          },
        ],
        id: 'e3653f73-e486-456b-9ba2-1e5421026175',
        isFavourite: false,
        patientSegmentReqiured: 123,
        projectName: 'test',
        userId: '0005a35c-b4b1-4fd4-bbd5-ab3e809e8b4e',
      },
    });
    await waitFor(() => {
      expect(ApiUtil.getData).toHaveBeenCalledWith(
        `${dashboardApiUrls.projectUrl}/e3653f73-e486-456b-9ba2-1e5421026175`,
      );
      expect(ApiUtil.getData).toHaveBeenCalledTimes(1);
      const { segments } = JSON.parse(
        JSON.stringify(Store.getState().feasibilityStudyReducer),
      );
      expect(segments.length).toBeGreaterThan(0);
    });
  });

  test('Render check', async () => {
    const data = {
      clientName: 'test',
      description: 'test test',
      feasibilityStudy: [
        {
          id: 'bfb1e873-755b-4ab7-9aa5-3ad4ec9a0dbb',
          projectId: 'e3653f73-e486-456b-9ba2-1e5421026175',
          isDone: false,
          feasibilityStudy: [
            {
              name: 'Indication',
              case: 'Indication',
              condition: 'Inclusion',
              caseName: 'Covid-19',
              child: [],
            },
            {
              name: 'Demographics',
              case: 'Demographics',
              condition: 'Inclusion',
              caseName: '',
              child: [
                {
                  condition: 'Gender',
                  field: 'gender',
                  gender: '',
                },
                {
                  condition: 'Age range',
                  field: 'age',
                  ageRange: {
                    min: 0,
                    max: 0,
                  },
                },
              ],
            },
          ],
          location: null,
          sitesContacted: null,
          sitesInitiated: null,
          preferredSites: null,
          sites: null,
          finalCount: 123,
        },
      ],
      id: 'e3653f73-e486-456b-9ba2-1e5421026175',
      isFavourite: false,
      patientSegmentReqiured: 123,
      projectName: 'test',
      userId: '0005a35c-b4b1-4fd4-bbd5-ab3e809e8b4e',
    };
    const { container } = renderWithContext(
      <ProjectLayout project={data} currentIndex={0} />,
    );
    const projectTitle = container.getElementsByClassName(
      'project-layout-indication-title',
    );
    const starElement = container.getElementsByClassName(
      'project-layout-favourite',
    );
    expect(projectTitle[0].textContent).toBe('Covid-19');
    userEvent.click(starElement[0]);
    const { projects } = JSON.parse(
      JSON.stringify(Store.getState().projectReducer),
    );
    expect(projects.length).toBeGreaterThan(0);
    expect(projects[0].isFavourite).toBe(true);
  });
  test(' icon', () => {
    const data = {
      clientName: 'test',
      description: 'test test',
      feasibilityStudy: [
        {
          id: 'bfb1e873-755b-4ab7-9aa5-3ad4ec9a0dbb',
          projectId: 'e3653f73-e486-456b-9ba2-1e5421026175',
          isDone: false,
          feasibilityStudy: [
            {
              name: 'Indication',
              case: 'Indication',
              condition: 'Inclusion',
              caseName: 'Covid-19',
              child: [],
            },
            {
              name: 'Demographics',
              case: 'Demographics',
              condition: 'Inclusion',
              caseName: '',
              child: [
                {
                  condition: 'Gender',
                  field: 'gender',
                  gender: '',
                },
                {
                  condition: 'Age range',
                  field: 'age',
                  ageRange: {
                    min: 0,
                    max: 0,
                  },
                },
              ],
            },
          ],
          location: null,
          sitesContacted: null,
          sitesInitiated: null,
          preferredSites: null,
          sites: null,
          finalCount: 123,
        },
      ],
      id: 'e3653f73-e486-456b-9ba2-1e5421026175',
      isFavourite: true,
      patientSegmentReqiured: 123,
      projectName: 'test',
      userId: '0005a35c-b4b1-4fd4-bbd5-ab3e809e8b4e',
    };
    const { container } = renderWithContext(
      <ProjectLayout project={data} currentIndex={0} />,
    );
    const starElement = container.getElementsByClassName(
      'project-layout-favourite',
    );
    userEvent.click(starElement[0]);
    const { projects } = JSON.parse(
      JSON.stringify(Store.getState().projectReducer),
    );
    expect(projects.length).toBeGreaterThan(0);
    expect(projects[0].isFavourite).toBe(false);
  });
});
