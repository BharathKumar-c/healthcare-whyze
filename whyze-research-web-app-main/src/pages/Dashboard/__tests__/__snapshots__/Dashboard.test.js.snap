// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test case for Dashbaord Dashboard render check 1`] = `
<div>
  <div
    id="dashboard"
  >
    <div
      class="dashboard"
    >
      <div
        class="dashboard-container"
      >
        <span
          class="ant-typography dashboard-title"
        >
          Welcome Anne!
        </span>
        <div
          class="dashboard-content"
        >
          <span
            class="ant-typography dashboard-content-text"
          >
            See how Whyze is preforming for researchers like you.
          </span>
        </div>
        <div
          class="ant-row"
        >
          <div
            class="ant-col ant-col-6 dashboard-detail-box"
          >
            <div
              class="ant-row ant-row-middle"
            >
              <div
                class="ant-col ant-col-5"
              >
                <div
                  class="dashboard-patient-icon"
                >
                  <svg>
                    PopulationVector.svg
                  </svg>
                </div>
              </div>
              <div
                class="ant-col ant-col-19 dashboard-patient-text"
              >
                Patients
              </div>
            </div>
            <div
              class="ant-row"
            >
              <div
                class="ant-col ant-col-5"
              />
              <div
                class="ant-col ant-col-19 dashboard-patient-container"
              >
                <div
                  class="dashboard-alignment"
                >
                  <div
                    class="dashboard-patient-total"
                    span="10"
                  >
                    455K
                  </div>
                  <div
                    class="dashboard-patient-total-text"
                    span="14"
                  >
                    Total patients
                  </div>
                </div>
                <div
                  class="dashboard-patient-grow-container"
                >
                  <svg>
                    ArrowUp.svg
                  </svg>
                  <span
                    class="ant-typography dashboard-patient-grow-percentage"
                  >
                    25%
                  </span>
                  <span
                    class="ant-typography dashboard-patient-grow-text"
                  >
                    monthly growth
                  </span>
                </div>
                <div
                  class="dashboard-alignment"
                >
                  <div
                    class="dashboard-patient-total"
                  >
                    35
                  </div>
                  <div
                    class="dashboard-patient-total-text"
                  >
                    Countries
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-col ant-col-6 dashboard-detail-box"
          >
            <div
              class="ant-row ant-row-middle"
            >
              <div
                class="ant-col ant-col-5"
              >
                <div
                  class="dashboard-sites-icon"
                >
                  <svg>
                    Sites.svg
                  </svg>
                </div>
              </div>
              <div
                class="ant-col ant-col-19 dashboard-patient-text"
              >
                Sites
              </div>
            </div>
            <div
              class="ant-row"
            >
              <div
                class="ant-col ant-col-5"
              />
              <div
                class="ant-col ant-col-19"
              >
                <div
                  class="dashboard-alignment"
                >
                  <div
                    class="dashboard-patient-total"
                  >
                    9000
                  </div>
                  <div
                    class="dashboard-patient-total-text"
                  >
                    Sites recruited
                  </div>
                </div>
                <div
                  class="dashboard-patient-grow-container"
                >
                  <svg>
                    ArrowUp.svg
                  </svg>
                  <span
                    class="ant-typography dashboard-patient-grow-percentage"
                  >
                    12%
                  </span>
                  <span
                    class="ant-typography dashboard-patient-grow-text"
                  >
                    monthly growth
                  </span>
                </div>
                <div
                  class="dashboard-alignment"
                >
                  <div
                    class="dashboard-patient-total"
                  >
                    1.5 wk
                  </div>
                  <div
                    class="dashboard-patient-total-text"
                  >
                    Avg. response time
                  </div>
                </div>
                <div
                  class="dashboard-patient-grow-container"
                >
                  <svg>
                    ArrowUp.svg
                  </svg>
                  <span
                    class="ant-typography dashboard-patient-grow-percentage"
                  >
                    20% faster
                  </span>
                  <span
                    class="ant-typography dashboard-patient-grow-text"
                  >
                    than industry
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-col ant-col-10 dashboard-detail-box"
          >
            <div
              class="ant-row ant-row-middle"
            >
              <div
                class="ant-col ant-col-3"
              >
                <div
                  class="dashboard-trials-icon"
                >
                  <svg>
                    TestTube.svg
                  </svg>
                </div>
              </div>
              <div
                class="ant-col ant-col-21 dashboard-patient-text"
              >
                Trials
              </div>
            </div>
            <div
              class="ant-row"
            >
              <div
                class="ant-col ant-col-3"
              />
              <div
                class="ant-col ant-col-21"
              >
                <div
                  class="dashboard-alignment-trail"
                >
                  <div
                    class="dashboard-trials-left"
                  >
                    <div
                      class="dashboard-alignment"
                    >
                      <div
                        class="dashboard-patient-total"
                      >
                        800
                      </div>
                      <div
                        class="dashboard-patient-total-text"
                      >
                        Active trials
                      </div>
                    </div>
                    <div
                      class="dashboard-patient-grow-container"
                    >
                      <svg>
                        ArrowUp.svg
                      </svg>
                      <span
                        class="ant-typography dashboard-patient-grow-percentage"
                      >
                        12%
                      </span>
                      <span
                        class="ant-typography dashboard-patient-grow-text"
                      >
                        monthly growth
                      </span>
                    </div>
                    <div
                      class="dashboard-alignment"
                    >
                      <div
                        class="dashboard-patient-total"
                      >
                        12K
                      </div>
                      <div
                        class="dashboard-patient-total-text"
                      >
                        Total trials
                      </div>
                    </div>
                    <div
                      class="dashboard-patient-grow-container-gray"
                    >
                      <span
                        class="ant-typography dashboard-patient-grow-container-gray-percentage"
                      >
                        In less than 4 years
                      </span>
                    </div>
                  </div>
                  <div
                    class="dashboard-vertical-divider-container"
                  >
                    <div
                      class="dashboard-vertical-divider"
                    />
                  </div>
                  <div
                    class="dashboard-trials-right"
                  >
                    <div>
                      <div
                        class="dashboard-alignment"
                      >
                        <div
                          class="dashboard-patient-total"
                        >
                          155
                        </div>
                        <div
                          class="dashboard-patient-total-text"
                        >
                          Screen fail rate
                        </div>
                      </div>
                      <div
                        class="dashboard-patient-grow-container"
                      >
                        <svg>
                          ArrowDown.svg
                        </svg>
                        <span
                          class="ant-typography dashboard-patient-grow-percentage"
                        >
                          20% lower
                        </span>
                        <span
                          class="ant-typography dashboard-patient-grow-text"
                        >
                          than industry
                        </span>
                      </div>
                      <div
                        class="dashboard-alignment"
                      >
                        <div
                          class="dashboard-patient-total"
                        >
                          3 months
                        </div>
                        <div
                          class="dashboard-patient-total-text"
                        >
                          Avg. time to recruit
                        </div>
                      </div>
                      <div
                        class="dashboard-patient-grow-container"
                      >
                        <svg>
                          ArrowUp.svg
                        </svg>
                        <span
                          class="ant-typography dashboard-patient-grow-percentage"
                        >
                          20% faster
                        </span>
                        <span
                          class="ant-typography dashboard-patient-grow-text"
                        >
                          than industry
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="dashboard-patient-study-container"
        >
          <div
            class="dashboard-patient-study-text"
          >
            <div>
              <span
                class="ant-typography dashboard-title"
              >
                Create your first study
              </span>
            </div>
            <div>
              <span
                class="ant-typography dashboard-patient-study-content"
              >
                Define your protocol or create your first feasibility study
              </span>
            </div>
          </div>
          <div
            class="dashboard-patient-study-btn"
          >
            <button
              class="ant-btn ant-btn-default dashboard-patient-study-button"
              data-testid="dashboard-patient-study-button"
              type="button"
            >
              <span>
                Let's go
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
